cmd/thv/app/mcp.go

```diff
@@ -0,0 +1,336 @@
+package app
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"text/tabwriter"
+	"time"
+
+	"github.com/mark3labs/mcp-go/client"
+	"github.com/mark3labs/mcp-go/mcp"
+	"github.com/spf13/cobra"
+
+	"github.com/stacklok/toolhive/pkg/logger"
+	"github.com/stacklok/toolhive/pkg/versions"
+)
+
+var (
+	mcpServerURL string
+	mcpFormat    string
+	mcpTimeout   time.Duration
+)
+
+func newMCPCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "mcp",
+		Short: "Interact with MCP servers for debugging",
+		Long:  `The mcp command provides subcommands to interact with MCP (Model Context Protocol) servers for debugging purposes.`,
+	}
+
+	// Create list command
+	listCmd := &cobra.Command{
+		Use:   "list [tools|resources|prompts]",
+		Short: "List MCP server capabilities",
+		Long:  `List tools, resources, and prompts available from an MCP server. Use subcommands to list specific types.`,
+		RunE:  mcpListCmdFunc,
+	}
+
+	// Create specific list subcommands
+	toolsCmd := &cobra.Command{
+		Use:   "tools",
+		Short: "List available tools from MCP server",
+		Long:  `List all tools available from the specified MCP server.`,
+		RunE:  mcpListToolsCmdFunc,
+	}
+
+	resourcesCmd := &cobra.Command{
+		Use:   "resources",
+		Short: "List available resources from MCP server",
+		Long:  `List all resources available from the specified MCP server.`,
+		RunE:  mcpListResourcesCmdFunc,
+	}
+
+	promptsCmd := &cobra.Command{
+		Use:   "prompts",
+		Short: "List available prompts from MCP server",
+		Long:  `List all prompts available from the specified MCP server.`,
+		RunE:  mcpListPromptsCmdFunc,
+	}
+
+	// Add flags to all MCP commands
+	addMCPFlags(listCmd)
+	addMCPFlags(toolsCmd)
+	addMCPFlags(resourcesCmd)
+	addMCPFlags(promptsCmd)
+
+	// Add specific list subcommands to list command
+	listCmd.AddCommand(toolsCmd)
+	listCmd.AddCommand(resourcesCmd)
+	listCmd.AddCommand(promptsCmd)
+
+	// Add list subcommand to mcp
+	cmd.AddCommand(listCmd)
+
+	return cmd
+}
+
+func addMCPFlags(cmd *cobra.Command) {
+	cmd.Flags().StringVar(&mcpServerURL, "server", "", "MCP server URL (required)")
+	cmd.Flags().StringVar(&mcpFormat, "format", FormatText, "Output format (json or text)")
+	cmd.Flags().DurationVar(&mcpTimeout, "timeout", 30*time.Second, "Connection timeout")
+	_ = cmd.MarkFlagRequired("server")
+}
+
+// mcpListCmdFunc lists all capabilities (tools, resources, prompts)
+func mcpListCmdFunc(cmd *cobra.Command, _ []string) error {
+	ctx, cancel := context.WithTimeout(cmd.Context(), mcpTimeout)
```

Comment:
Let's follow a similar pattern we have used for other functionality where the operations are implemented behind an interface, and the CLI calls them. This will save us trouble when we need to share them elsewhere.

Reply:
Where would we share them? I don't see the use case